### PR TITLE
treewide: drop poetry2nix

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

poetry2nix in-tree was broken in a pretty major way for a while now, and the only remaining consumers in nixpkgs itself were NixOps versions. So, drop NixOps stable (it's on Python 2, which is very dead), rework NixOps unstable to use nixpkgs dependencies, and finally drop poetry2nix itself.

I've taken the liberty to add upstream maintainers as maintainers to the respective NixOps plugins - if you'd like to be excluded, please say so.

## TODO
- [ ] collect new maintainer signoffs
  - [ ] @Kiwi for nixops-digitalocean
  - [x] @lukebfox for nixops-hetznercloud
- [ ] merge this

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by infinisil** [comment]
How's the in-tree poetry2nix broken? And why can't we just update it to fix it?

**Comment by K900** [comment]
It was broken by the Python bootstrap changes, and we don't really want to update it because the out-of-tree version is generally better, because it has a lot more overrides and adds them a lot faster. We also don't want to encourage new poetry2nix consumers in-tree.

File: pkgs/top-level/aliases.nix
Comment (infinisil) [inline pkgs/top-level/aliases.nix:600]: Should be an error instead, otherwise we'd be updating all nixops users to unstable eithout warning.

But then also it would be more proper to just deprecate stable nixops softly with a warning first for a release.

Comment (K900) [reply pkgs/top-level/aliases.nix:600]: It's been failing to eval for months now due to Python 2 insecure stuff.

Comment (K900) [reply pkgs/top-level/aliases.nix]: Updated to a throw.

Comment (infinisil) [reply pkgs/top-level/aliases.nix]: Is it usable in the stable release? If so, we need to deprecate it with a warning or error

Comment (K900) [reply pkgs/top-level/aliases.nix]: It's not, Python 2.7 is marked insecure in 23.05 as well.

**Review by RaitoBezarius — APPROVED** [review]
This is a release manager seal :heavy_check_mark:.

File: pkgs/top-level/aliases.nix
Comment (infinisil) [reply pkgs/top-level/aliases.nix]: Wouldn't somebody running into that just ignore the warning?

Comment (K900) [reply pkgs/top-level/aliases.nix]: I would hope not, given 2.7 was EOL in _2020_.

Comment (infinisil) [reply pkgs/top-level/aliases.nix]: Still, let's have at least an error here. People should be made aware that they're using the stable attribute and should switch to the unstable one

Comment (K900) [reply pkgs/top-level/aliases.nix]: I did make it a throw.

**Comment by K900** [comment]
Marking as draft to prevent accidental merges.

**Comment by K900** [comment]
Fixed eval error.

File: pkgs/top-level/aliases.nix
Comment (chayleaf) [inline pkgs/top-level/aliases.nix:700]:
```suggestion
  poetry2nix = throw "poetry2nix is now maintained out-of-tree. Please use https://github.com/nix-community/poetry2nix/"; # Added 2023-10-26
```

File: pkgs/top-level/aliases.nix
Comment (chayleaf) [inline pkgs/top-level/aliases.nix:600]:
```suggestion
  nixops = throw "'nixops' has been removed. Please use 'nixops_unstable' for the time being."; # Added 2023-10-26
```

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (chayleaf) [inline pkgs/development/python-modules/digitalocean/default.nix:43]: maybe add a comment why the test is disabled?

File: pkgs/top-level/aliases.nix
Comment (K900) [reply pkgs/top-level/aliases.nix]: Thanks, done.

File: pkgs/top-level/aliases.nix
Comment (K900) [reply pkgs/top-level/aliases.nix]: Done.

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (K900) [reply pkgs/development/python-modules/digitalocean/default.nix:43]: Done.

File: pkgs/applications/networking/cluster/nixops/default.nix
Comment (dotlambda) [inline pkgs/applications/networking/cluster/nixops/default.nix:29]:
```suggestion
  # e.g. nixops_unstable.withPlugins (ps: with ps; [ nixops-aws ])
```

File: pkgs/applications/networking/cluster/nixops/default.nix
Comment (dotlambda) [inline pkgs/applications/networking/cluster/nixops/default.nix:55]: What's the reasoning behind adding these nine by default rather than all of them?

File: pkgs/applications/networking/cluster/nixops/plugins/nixops-libvirtd.nix
Comment (dotlambda) [inline pkgs/applications/networking/cluster/nixops/plugins/nixops-libvirtd.nix:42]:
```suggestion
    description = "NixOps libvirtd backend plugin";
```

File: pkgs/applications/networking/cluster/nixops/plugins/nixops-libvirtd.nix
Comment (dotlambda) [inline pkgs/applications/networking/cluster/nixops/plugins/nixops-libvirtd.nix:44]: Which one exactly?

File: pkgs/applications/networking/cluster/nixops/plugins/nixops-vbox.nix
Comment (dotlambda) [inline pkgs/applications/networking/cluster/nixops/plugins/nixops-vbox.nix:37]: Shouldn't be empty.

File: pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix
Comment (dotlambda) [inline pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix:37]: Shouldn't be empty.

File: pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix
Comment (dotlambda) [inline pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix:40]: Maintainers missing.

File: pkgs/applications/networking/cluster/nixops/unwrapped.nix
Comment (dotlambda) [inline pkgs/applications/networking/cluster/nixops/unwrapped.nix:64]:
```suggestion
    description = "Tool for deploying to NixOS machines in a network or cloud";
```

File: pkgs/applications/networking/cluster/nixops/default.nix
Comment (K900) [reply pkgs/applications/networking/cluster/nixops/default.nix:55]: That's what the previous version of the expression did.

File: pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix
Comment (K900) [reply pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix:40]: I'm not really sure what to put there, because the project hasn't been touched since 2021 and has no maintainers declared. Any volunteers? :)

File: pkgs/applications/networking/cluster/nixops/default.nix
Comment (K900) [reply pkgs/applications/networking/cluster/nixops/default.nix:29]: Thanks, applied.

File: pkgs/applications/networking/cluster/nixops/plugins/nixops-libvirtd.nix
Comment (K900) [reply pkgs/applications/networking/cluster/nixops/plugins/nixops-libvirtd.nix:42]: Huh, I thought I removed this. Anyway, fixed.

File: pkgs/applications/networking/cluster/nixops/plugins/nixops-libvirtd.nix
Comment (K900) [reply pkgs/applications/networking/cluster/nixops/plugins/nixops-libvirtd.nix:44]: Fixed.

File: pkgs/applications/networking/cluster/nixops/unwrapped.nix
Comment (K900) [reply pkgs/applications/networking/cluster/nixops/unwrapped.nix:64]: Fixed.

File: pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix
Comment (K900) [reply pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix:37]: Fixed.

File: pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix
Comment (RaitoBezarius) [reply pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix:40]: empty maintainers if not imho

Comment (K900) [reply pkgs/applications/networking/cluster/nixops/plugins/nixos-modules-contrib.nix:44]: Done.

**Comment by K900** [comment]
Updated to latest version of nixops with all the fixes, also added updateScript everywhere.

**Comment by K900** [comment]
This should be good to go now, though I'd prefer to get explicit agreement from people added as maintainers before we merge.

File: pkgs/top-level/aliases.nix
Comment (dotlambda) [inline pkgs/top-level/aliases.nix:600]:
```suggestion
  nixops = throw "'nixops' has been removed. Please use 'nixops_unstable' for the time being."; # Added 2023-10-26
  nixops-dns = throw "...";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/top-level/aliases.nix
Comment (K900) [reply pkgs/top-level/aliases.nix:600]: Actually, maybe we can keep nixops-dns? I'm not sure if the state file format has changed.

Comment (K900) [reply pkgs/top-level/aliases.nix:600]: At a glance, the state format has not changed, so it's OK to keep. I've moved it to `pkgs/by-name` though, because now it definitely has no business being in package-management.

**Review by dotlambda — APPROVED** [review]

**Review by sorki — APPROVED** [review]

**Comment by lukebfox** [comment]
Thanks for waiting, I'm happy to maintain my plugin's nix-expression here.

**Comment by K900** [comment]
I'm just going to merge this, and if @Kiwi wants to step down, they're welcome to. I'm really worried about this bitrotting.

**Pull request merged**
